*** ./smali/com/android/server/power/PowerManagerService.smali	2015-05-02 01:21:32.324407094 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/services.jar.out/./smali/com/android/server/power/PowerManagerService.smali	2015-05-02 01:17:58.628413408 +0800
***************
*** 1258,1263 ****
--- 1258,1265 ----
  
      const/4 v5, 0x0
  
+     invoke-direct {p0}, Lcom/android/server/power/PowerManagerService;->checkIfBootAnimationFinished()V
+ 
      iget-wide v6, p0, Lcom/android/server/power/PowerManagerService;->mLastWakeTime:J
  
      cmp-long v6, p1, v6
***************
*** 3270,3275 ****
--- 3272,3283 ----
      .end local v2    # "screenAutoBrightnessAdjustment":F
      .end local v3    # "screenBrightness":I
      :cond_4
+     invoke-direct {p0}, Lcom/android/server/power/PowerManagerService;->getDesiredScreenPowerStateLocked()I
+ 
+     move-result v6
+ 
+     invoke-static {v6}, Lcom/android/server/power/ButtonLightController;->turnOffButtonLight(I)V
+ 
      return-void
  
      .restart local v1    # "newScreenState":I
***************
*** 4385,4390 ****
--- 4393,4400 ----
  
      const/4 v1, 0x0
  
+     invoke-static {}, Lcom/android/server/power/ButtonLightController;->setButtonLightTimeout()V
+ 
      iget-wide v2, p0, Lcom/android/server/power/PowerManagerService;->mLastSleepTime:J
  
      cmp-long v2, p1, v2
***************
*** 7470,7475 ****
--- 7480,7507 ----
      invoke-direct/range {p0 .. p0}, Lcom/android/server/power/PowerManagerService;->updatePowerStateLocked()V
  
      monitor-exit v18
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     move-object/from16 v0, p0
+ 
+     iget-object v2, v0, Lcom/android/server/power/PowerManagerService;->mContext:Landroid/content/Context;
+ 
+     move-object/from16 v0, p0
+ 
+     iget-object v3, v0, Lcom/android/server/power/PowerManagerService;->mHandler:Lcom/android/server/power/PowerManagerService$PowerManagerHandler;
+ 
+     move-object/from16 v0, p0
+ 
+     iget-object v4, v0, Lcom/android/server/power/PowerManagerService;->mLightsService:Lcom/android/server/LightsService;
+ 
+     const/4 v5, 0x2
+ 
+     invoke-virtual {v4, v5}, Lcom/android/server/LightsService;->getLight(I)Lcom/android/server/LightsService$Light;
+ 
+     move-result-object v4
+ 
+     invoke-static {v2, v3, v4}, Lcom/android/server/power/ButtonLightController;->setButtonLight(Landroid/content/Context;Landroid/os/Handler;Lcom/android/server/LightsService$Light;)V
  
      return-void
  
***************
*** 7480,7488 ****
      :catchall_0
      move-exception v2
  
      monitor-exit v18
!     :try_end_0
!     .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      throw v2
  .end method
--- 7512,7521 ----
      :catchall_0
      move-exception v2
  
+     :try_start_1
      monitor-exit v18
!     :try_end_1
!     .catchall {:try_start_1 .. :try_end_1} :catchall_0
  
      throw v2
  .end method
