*** ./smali/com/android/internal/widget/ILockSettings$Stub.smali	2015-05-02 01:21:50.896406545 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework2.jar.out/./smali/com/android/internal/widget/ILockSettings$Stub.smali	2015-05-02 01:18:17.848412840 +0800
***************
*** 50,55 ****
--- 50,57 ----
  
  .field static final TRANSACTION_setLong:I = 0x2
  
+ .field static final TRANSACTION_setRawLockPassword:I = 0xe
+ 
  .field static final TRANSACTION_setString:I = 0x3
  
  
***************
*** 551,556 ****
--- 553,579 ----
  
      goto/16 :goto_0
  
+     :sswitch_miui_0
+     const-string v7, "com.android.internal.widget.ILockSettings"
+ 
+     invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
+ 
+     invoke-virtual {p2}, Landroid/os/Parcel;->createByteArray()[B
+ 
+     move-result-object v0
+ 
+     .local v0, "_arg0":[B
+     invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v1
+ 
+     .restart local v1    # "_arg1":I
+     invoke-virtual {p0, v0, v1}, Lcom/android/internal/widget/ILockSettings$Stub;->setRawLockPassword([BI)V
+ 
+     invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
+ 
+     goto/16 :goto_0
+ 
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_1
***************
*** 566,571 ****
--- 589,595 ----
          0xb -> :sswitch_b
          0xc -> :sswitch_c
          0xd -> :sswitch_d
+         0xe -> :sswitch_miui_0
          0x5f4e5446 -> :sswitch_0
      .end sparse-switch
  .end method
