*** ./smali/com/android/internal/telephony/uicc/UiccCard.smali	2015-05-02 01:21:56.060406393 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/uicc/UiccCard.smali	2015-05-02 01:18:23.168412683 +0800
***************
*** 313,318 ****
--- 313,525 ----
  .end method
  
  .method private onIccSwap(Z)V
+     .locals 14
+     .param p1, "isAdded"    # Z
+ 
+     .prologue
+     const/4 v3, 0x1
+ 
+     .local v3, "isHotSwapSupported":Z
+     const/4 v5, 0x0
+ 
+     .local v5, "needReboot":Z
+     iget-object v11, p0, Lcom/android/internal/telephony/uicc/UiccCard;->mLock:Ljava/lang/Object;
+ 
+     monitor-enter v11
+ 
+     :try_start_0
+     invoke-static {}, Landroid/content/res/Resources;->getSystem()Landroid/content/res/Resources;
+ 
+     move-result-object v7
+ 
+     .local v7, "r":Landroid/content/res/Resources;
+     const-string v9, ""
+ 
+     .local v9, "title":Ljava/lang/String;
+     const/4 v4, 0x0
+ 
+     .local v4, "listener":Landroid/content/DialogInterface$OnClickListener;
+     sget v6, Lcom/android/internal/R$string;->ok:I
+ 
+     .local v6, "positiveButtonText":I
+     if-eqz v3, :cond_5
+ 
+     if-eqz p1, :cond_4
+ 
+     sget v10, Lcom/android/internal/R$string;->sim_added_title:I
+ 
+     invoke-virtual {v7, v10}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
+ 
+     move-result-object v9
+ 
+     :goto_0
+     if-eqz v5, :cond_0
+ 
+     sget v6, Landroid/miui/R$string;->sim_added_reboot_confirm:I
+ 
+     new-instance v4, Lcom/android/internal/telephony/uicc/UiccCard$1;
+ 
+     .end local v4    # "listener":Landroid/content/DialogInterface$OnClickListener;
+     invoke-direct {v4, p0}, Lcom/android/internal/telephony/uicc/UiccCard$1;-><init>(Lcom/android/internal/telephony/uicc/UiccCard;)V
+ 
+     .restart local v4    # "listener":Landroid/content/DialogInterface$OnClickListener;
+     :cond_0
+     new-instance v10, Landroid/app/AlertDialog$Builder;
+ 
+     iget-object v12, p0, Lcom/android/internal/telephony/uicc/UiccCard;->mContext:Landroid/content/Context;
+ 
+     sget v13, Lmiui/R$style;->Theme_Light_Dialog_Alert:I
+ 
+     invoke-direct {v10, v12, v13}, Landroid/app/AlertDialog$Builder;-><init>(Landroid/content/Context;I)V
+ 
+     invoke-virtual {v10, v9}, Landroid/app/AlertDialog$Builder;->setTitle(Ljava/lang/CharSequence;)Landroid/app/AlertDialog$Builder;
+ 
+     move-result-object v10
+ 
+     invoke-virtual {v10, v6, v4}, Landroid/app/AlertDialog$Builder;->setPositiveButton(ILandroid/content/DialogInterface$OnClickListener;)Landroid/app/AlertDialog$Builder;
+ 
+     move-result-object v0
+ 
+     .local v0, "builder":Landroid/app/AlertDialog$Builder;
+     if-eqz v5, :cond_1
+ 
+     sget v10, Lcom/android/internal/R$string;->cancel:I
+ 
+     const/4 v12, 0x0
+ 
+     invoke-virtual {v0, v10, v12}, Landroid/app/AlertDialog$Builder;->setNegativeButton(ILandroid/content/DialogInterface$OnClickListener;)Landroid/app/AlertDialog$Builder;
+ 
+     :cond_1
+     invoke-virtual {v0}, Landroid/app/AlertDialog$Builder;->create()Landroid/app/AlertDialog;
+ 
+     move-result-object v1
+ 
+     .local v1, "dialog":Landroid/app/AlertDialog;
+     invoke-virtual {v1}, Landroid/app/AlertDialog;->getWindow()Landroid/view/Window;
+ 
+     move-result-object v10
+ 
+     const/16 v12, 0x7d3
+ 
+     invoke-virtual {v10, v12}, Landroid/view/Window;->setType(I)V
+ 
+     if-nez p1, :cond_2
+ 
+     invoke-virtual {v1}, Landroid/app/AlertDialog;->show()V
+ 
+     :cond_2
+     if-nez v5, :cond_3
+ 
+     new-instance v8, Ljava/util/Timer;
+ 
+     invoke-direct {v8}, Ljava/util/Timer;-><init>()V
+ 
+     .local v8, "t":Ljava/util/Timer;
+     new-instance v10, Lcom/android/internal/telephony/uicc/UiccCard$TimerTask;
+ 
+     invoke-direct {v10, p0, v1, v8}, Lcom/android/internal/telephony/uicc/UiccCard$TimerTask;-><init>(Lcom/android/internal/telephony/uicc/UiccCard;Landroid/app/AlertDialog;Ljava/util/Timer;)V
+ 
+     const-wide/16 v12, 0x2710
+ 
+     invoke-virtual {v8, v10, v12, v13}, Ljava/util/Timer;->schedule(Ljava/util/TimerTask;J)V
+ 
+     new-instance v2, Landroid/content/Intent;
+ 
+     const-string v10, "android.intent.action.SIM_HOT_SWAP"
+ 
+     invoke-direct {v2, v10}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
+ 
+     .local v2, "intent":Landroid/content/Intent;
+     const/high16 v10, 0x20000000
+ 
+     invoke-virtual {v2, v10}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
+ 
+     const-string v10, "is_sim_added"
+ 
+     invoke-virtual {v2, v10, p1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
+ 
+     new-instance v10, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v10}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v12, "Broadcasting intent ACTION_SIM_HOT_SWAP  isAdded: "
+ 
+     invoke-virtual {v10, v12}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v10
+ 
+     invoke-virtual {v10, p1}, Ljava/lang/StringBuilder;->append(Z)Ljava/lang/StringBuilder;
+ 
+     move-result-object v10
+ 
+     invoke-virtual {v10}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v10
+ 
+     invoke-direct {p0, v10}, Lcom/android/internal/telephony/uicc/UiccCard;->log(Ljava/lang/String;)V
+ 
+     const-string v10, "android.permission.READ_PHONE_STATE"
+ 
+     const/4 v12, -0x1
+ 
+     invoke-static {v2, v10, v12}, Landroid/app/ActivityManagerNative;->broadcastStickyIntent(Landroid/content/Intent;Ljava/lang/String;I)V
+ 
+     .end local v2    # "intent":Landroid/content/Intent;
+     .end local v8    # "t":Ljava/util/Timer;
+     :cond_3
+     monitor-exit v11
+ 
+     return-void
+ 
+     .end local v0    # "builder":Landroid/app/AlertDialog$Builder;
+     .end local v1    # "dialog":Landroid/app/AlertDialog;
+     :cond_4
+     sget v10, Lcom/android/internal/R$string;->sim_removed_title:I
+ 
+     invoke-virtual {v7, v10}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
+ 
+     move-result-object v9
+ 
+     goto :goto_0
+ 
+     :cond_5
+     if-eqz p1, :cond_6
+ 
+     sget v10, Landroid/miui/R$string;->sim_added_reboot_title:I
+ 
+     invoke-virtual {v7, v10}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
+ 
+     move-result-object v9
+ 
+     :goto_1
+     move v5, p1
+ 
+     goto/16 :goto_0
+ 
+     :cond_6
+     sget v10, Lcom/android/internal/R$string;->sim_removed_title:I
+ 
+     invoke-virtual {v7, v10}, Landroid/content/res/Resources;->getString(I)Ljava/lang/String;
+ 
+     move-result-object v9
+ 
+     goto :goto_1
+ 
+     .end local v4    # "listener":Landroid/content/DialogInterface$OnClickListener;
+     .end local v6    # "positiveButtonText":I
+     .end local v7    # "r":Landroid/content/res/Resources;
+     .end local v9    # "title":Ljava/lang/String;
+     :catchall_0
+     move-exception v10
+ 
+     monitor-exit v11
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     throw v10
+ .end method
+ 
+ .method private onIccSwap_aosp(Z)V
      .locals 10
      .param p1, "isAdded"    # Z
  
***************
*** 1241,1246 ****
--- 1448,1489 ----
      throw v0
  .end method
  
+ .method public getNumApplications()I
+     .locals 5
+ 
+     .prologue
+     const/4 v2, 0x0
+ 
+     .local v2, "count":I
+     iget-object v1, p0, Lcom/android/internal/telephony/uicc/UiccCard;->mUiccApplications:[Lcom/android/internal/telephony/uicc/UiccCardApplication;
+ 
+     .local v1, "arr$":[Lcom/android/internal/telephony/uicc/UiccCardApplication;
+     array-length v4, v1
+ 
+     .local v4, "len$":I
+     const/4 v3, 0x0
+ 
+     .local v3, "i$":I
+     :goto_0
+     if-ge v3, v4, :cond_1
+ 
+     aget-object v0, v1, v3
+ 
+     .local v0, "a":Lcom/android/internal/telephony/uicc/UiccCardApplication;
+     if-eqz v0, :cond_0
+ 
+     add-int/lit8 v2, v2, 0x1
+ 
+     :cond_0
+     add-int/lit8 v3, v3, 0x1
+ 
+     goto :goto_0
+ 
+     .end local v0    # "a":Lcom/android/internal/telephony/uicc/UiccCardApplication;
+     :cond_1
+     return v2
+ .end method
+ 
  .method public getUniversalPinState()Lcom/android/internal/telephony/uicc/IccCardStatus$PinState;
      .locals 2
  
