*** ./smali/android/media/AudioService$VolumeStreamState.smali	2015-05-02 01:20:56.764408145 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework.jar.out/./smali/android/media/AudioService$VolumeStreamState.smali	2015-05-02 01:17:19.748414557 +0800
***************
*** 514,519 ****
--- 514,547 ----
      .param p2, "device"    # I
  
      .prologue
+     iget v0, p0, Landroid/media/AudioService$VolumeStreamState;->mStreamType:I
+ 
+     if-nez v0, :cond_miui_0
+ 
+     iget-object v0, p0, Landroid/media/AudioService$VolumeStreamState;->this$0:Landroid/media/AudioService;
+ 
+     # getter for: Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
+     invoke-static {v0}, Landroid/media/AudioService;->access_mVolumePanel(Landroid/media/AudioService;)Lmiui/view/VolumePanel;
+ 
+     move-result-object v0
+ 
+     invoke-virtual {v0}, Lmiui/view/VolumePanel;->isVoiceMutable()Z
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_miui_0
+ 
+     invoke-virtual {p0, p2}, Landroid/media/AudioService$VolumeStreamState;->getIndex(I)I
+ 
+     move-result v0
+ 
+     add-int/2addr v0, p1
+ 
+     if-gtz v0, :cond_miui_0
+ 
+     const/4 p1, 0x0
+ 
+     :cond_miui_0
      invoke-virtual {p0, p2}, Landroid/media/AudioService$VolumeStreamState;->getIndex(I)I
  
      move-result v0
***************
*** 1450,1455 ****
--- 1478,1513 ----
  
      move-result p1
  
+     if-nez p1, :cond_miui_0
+ 
+     iget v7, p0, Landroid/media/AudioService$VolumeStreamState;->mStreamType:I
+ 
+     if-nez v7, :cond_miui_0
+ 
+     iget-object v7, p0, Landroid/media/AudioService$VolumeStreamState;->this$0:Landroid/media/AudioService;
+ 
+     # getter for: Landroid/media/AudioService;->mVolumePanel:Lmiui/view/VolumePanel;
+     invoke-static {v7}, Landroid/media/AudioService;->access_mVolumePanel(Landroid/media/AudioService;)Lmiui/view/VolumePanel;
+ 
+     move-result-object v7
+ 
+     invoke-virtual {v7}, Lmiui/view/VolumePanel;->isVoiceMutable()Z
+ 
+     move-result v7
+ 
+     if-eqz v7, :cond_miui_0
+ 
+     invoke-static {}, Landroid/os/Binder;->getCallingUid()I
+ 
+     move-result v7
+ 
+     const/16 v8, 0x3e9
+ 
+     if-eq v7, v8, :cond_miui_0
+ 
+     const/16 p1, 0xa
+ 
+     :cond_miui_0
      iget-object v7, p0, Landroid/media/AudioService$VolumeStreamState;->this$0:Landroid/media/AudioService;
  
      # getter for: Landroid/media/AudioService;->mCameraSoundForced:Ljava/lang/Boolean;
