*** ./smali/android/content/Context.smali	2015-05-02 01:20:38.996408670 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework.jar.out/./smali/android/content/Context.smali	2015-05-02 01:17:00.916415113 +0800
***************
*** 166,171 ****
--- 166,190 ----
  .method public abstract bindService(Landroid/content/Intent;Landroid/content/ServiceConnection;I)Z
  .end method
  
+ .method public bindService(Landroid/content/Intent;Landroid/content/ServiceConnection;II)Z
+     .locals 1
+     .param p1, "service"    # Landroid/content/Intent;
+     .param p2, "conn"    # Landroid/content/ServiceConnection;
+     .param p3, "flags"    # I
+     .param p4, "userHandle"    # I
+ 
+     .prologue
+     new-instance v0, Landroid/os/UserHandle;
+ 
+     invoke-direct {v0, p4}, Landroid/os/UserHandle;-><init>(I)V
+ 
+     invoke-virtual {p0, p1, p2, p3, v0}, Landroid/content/Context;->bindServiceAsUser(Landroid/content/Intent;Landroid/content/ServiceConnection;ILandroid/os/UserHandle;)Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method public bindServiceAsUser(Landroid/content/Intent;Landroid/content/ServiceConnection;ILandroid/os/UserHandle;)Z
      .locals 2
      .param p1, "service"    # Landroid/content/Intent;
