*** ./smali/com/android/internal/app/IAppOpsService$Stub$Proxy.smali	2015-05-02 01:21:46.036406689 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework2.jar.out/./smali/com/android/internal/app/IAppOpsService$Stub$Proxy.smali	2015-05-02 01:18:12.964412984 +0800
***************
*** 112,117 ****
--- 112,184 ----
      throw v3
  .end method
  
+ .method public checkOperationInternal(IILjava/lang/String;)I
+     .locals 6
+     .param p1, "code"    # I
+     .param p2, "uid"    # I
+     .param p3, "packageName"    # Ljava/lang/String;
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ 
+     .prologue
+     invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
+ 
+     move-result-object v0
+ 
+     .local v0, "_data":Landroid/os/Parcel;
+     invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
+ 
+     move-result-object v1
+ 
+     .local v1, "_reply":Landroid/os/Parcel;
+     :try_start_0
+     const-string v3, "com.android.internal.app.IAppOpsService"
+ 
+     invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
+ 
+     invoke-virtual {v0, p1}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     invoke-virtual {v0, p2}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     invoke-virtual {v0, p3}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
+ 
+     iget-object v3, p0, Lcom/android/internal/app/IAppOpsService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
+ 
+     const/16 v4, 0xe
+ 
+     const/4 v5, 0x0
+ 
+     invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     move-result v2
+ 
+     .local v2, "_result":I
+     invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
+ 
+     invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
+ 
+     return v2
+ 
+     .end local v2    # "_result":I
+     :catchall_0
+     move-exception v3
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
+ 
+     invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
+ 
+     throw v3
+ .end method
+ 
  .method public checkPackage(ILjava/lang/String;)I
      .locals 6
      .param p1, "uid"    # I
***************
*** 524,529 ****
--- 591,669 ----
      :catchall_0
      move-exception v3
  
+     invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
+ 
+     invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
+ 
+     throw v3
+ .end method
+ 
+ .method public registerCallback(Lcom/android/internal/app/IOpsCallback;)I
+     .locals 6
+     .param p1, "callback"    # Lcom/android/internal/app/IOpsCallback;
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ 
+     .prologue
+     invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
+ 
+     move-result-object v0
+ 
+     .local v0, "_data":Landroid/os/Parcel;
+     invoke-static {}, Landroid/os/Parcel;->obtain()Landroid/os/Parcel;
+ 
+     move-result-object v1
+ 
+     .local v1, "_reply":Landroid/os/Parcel;
+     :try_start_0
+     const-string v3, "com.android.internal.app.IAppOpsService"
+ 
+     invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeInterfaceToken(Ljava/lang/String;)V
+ 
+     if-eqz p1, :cond_0
+ 
+     invoke-interface {p1}, Lcom/android/internal/app/IOpsCallback;->asBinder()Landroid/os/IBinder;
+ 
+     move-result-object v3
+ 
+     :goto_0
+     invoke-virtual {v0, v3}, Landroid/os/Parcel;->writeStrongBinder(Landroid/os/IBinder;)V
+ 
+     iget-object v3, p0, Lcom/android/internal/app/IAppOpsService$Stub$Proxy;->mRemote:Landroid/os/IBinder;
+ 
+     const/16 v4, 0xd
+ 
+     const/4 v5, 0x0
+ 
+     invoke-interface {v3, v4, v0, v1, v5}, Landroid/os/IBinder;->transact(ILandroid/os/Parcel;Landroid/os/Parcel;I)Z
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->readException()V
+ 
+     invoke-virtual {v1}, Landroid/os/Parcel;->readInt()I
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     move-result v2
+ 
+     .local v2, "_result":I
+     invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
+ 
+     invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
+ 
+     return v2
+ 
+     .end local v2    # "_result":I
+     :cond_0
+     const/4 v3, 0x0
+ 
+     goto :goto_0
+ 
+     :catchall_0
+     move-exception v3
+ 
      invoke-virtual {v1}, Landroid/os/Parcel;->recycle()V
  
      invoke-virtual {v0}, Landroid/os/Parcel;->recycle()V
