*** ./smali/android/view/ScaleGestureDetector.smali	2015-05-02 01:21:08.792407789 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework.jar.out/./smali/android/view/ScaleGestureDetector.smali	2015-05-02 01:17:33.068414163 +0800
***************
*** 99,105 ****
  .end method
  
  .method public constructor <init>(Landroid/content/Context;Landroid/view/ScaleGestureDetector$OnScaleGestureListener;Landroid/os/Handler;)V
!     .locals 3
      .param p1, "context"    # Landroid/content/Context;
      .param p2, "listener"    # Landroid/view/ScaleGestureDetector$OnScaleGestureListener;
      .param p3, "handler"    # Landroid/os/Handler;
--- 99,105 ----
  .end method
  
  .method public constructor <init>(Landroid/content/Context;Landroid/view/ScaleGestureDetector$OnScaleGestureListener;Landroid/os/Handler;)V
!     .locals 4
      .param p1, "context"    # Landroid/content/Context;
      .param p2, "listener"    # Landroid/view/ScaleGestureDetector$OnScaleGestureListener;
      .param p3, "handler"    # Landroid/os/Handler;
***************
*** 107,112 ****
--- 107,114 ----
      .prologue
      const/4 v2, 0x0
  
+     const/high16 v3, 0x7fc00000    # NaNf
+ 
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      iput v2, p0, Landroid/view/ScaleGestureDetector;->mDoubleTapMode:I
***************
*** 178,183 ****
--- 180,191 ----
      invoke-virtual {p0, v1}, Landroid/view/ScaleGestureDetector;->setQuickScaleEnabled(Z)V
  
      :cond_0
+     iput v3, p0, Landroid/view/ScaleGestureDetector;->mTouchUpper:F
+ 
+     iput v3, p0, Landroid/view/ScaleGestureDetector;->mTouchLower:F
+ 
+     iput v3, p0, Landroid/view/ScaleGestureDetector;->mTouchHistoryLastAccepted:F
+ 
      return-void
  
      .end local v0    # "res":Landroid/content/res/Resources;
