*** ./smali/android/view/ViewDebug.smali	2015-05-02 01:21:11.788407701 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework.jar.out/./smali/android/view/ViewDebug.smali	2015-05-02 01:17:36.704414056 +0800
***************
*** 4688,4693 ****
--- 4688,4701 ----
      const-string v3, "\\n"
  
      invoke-virtual {v1, v2, v3}, Ljava/lang/String;->replace(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     const-string v2, "\r"
+ 
+     const-string v3, "\\r"
+ 
+     invoke-virtual {v1, v2, v3}, Ljava/lang/String;->replace(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Ljava/lang/String;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
