*** ./smali/com/android/server/am/ActivityStack.smali	2015-05-02 01:21:41.756406815 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/services.jar.out/./smali/com/android/server/am/ActivityStack.smali	2015-05-02 01:18:08.592413114 +0800
***************
*** 1560,1565 ****
--- 1560,1639 ----
      return-void
  .end method
  
+ .method private startPausingLocked_Middle(Lcom/android/server/am/ActivityRecord;)V
+     .locals 2
+     .param p1, "prev"    # Lcom/android/server/am/ActivityRecord;
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/am/ActivityStack;->mWindowManager:Lcom/android/server/wm/WindowManagerService;
+ 
+     invoke-virtual {v0}, Lcom/android/server/wm/WindowManagerService;->getFocusedWindowType()I
+ 
+     move-result v0
+ 
+     const/16 v1, 0x7d0
+ 
+     if-eq v0, v1, :cond_1
+ 
+     iget-object v0, p1, Lcom/android/server/am/ActivityRecord;->task:Lcom/android/server/am/TaskRecord;
+ 
+     iget-object v0, v0, Lcom/android/server/am/TaskRecord;->intent:Landroid/content/Intent;
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v0, p1, Lcom/android/server/am/ActivityRecord;->task:Lcom/android/server/am/TaskRecord;
+ 
+     iget-object v0, v0, Lcom/android/server/am/TaskRecord;->intent:Landroid/content/Intent;
+ 
+     invoke-virtual {v0}, Landroid/content/Intent;->getFlags()I
+ 
+     move-result v0
+ 
+     const/high16 v1, 0x800000
+ 
+     and-int/2addr v0, v1
+ 
+     if-nez v0, :cond_1
+ 
+     :cond_0
+     iget-object v1, p0, Lcom/android/server/am/ActivityStack;->mContext:Landroid/content/Context;
+ 
+     iget-object v0, p1, Lcom/android/server/am/ActivityRecord;->task:Lcom/android/server/am/TaskRecord;
+ 
+     iget-object v0, v0, Lcom/android/server/am/TaskRecord;->origActivity:Landroid/content/ComponentName;
+ 
+     if-eqz v0, :cond_2
+ 
+     iget-object v0, p1, Lcom/android/server/am/ActivityRecord;->task:Lcom/android/server/am/TaskRecord;
+ 
+     iget-object v0, v0, Lcom/android/server/am/TaskRecord;->origActivity:Landroid/content/ComponentName;
+ 
+     invoke-virtual {v0}, Landroid/content/ComponentName;->flattenToShortString()Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     :goto_0
+     invoke-static {v1, v0}, Lmiui/util/ScreenshotUtils;->captureActivityScreenshot(Landroid/content/Context;Ljava/lang/String;)V
+ 
+     :cond_1
+     return-void
+ 
+     :cond_2
+     iget-object v0, p1, Lcom/android/server/am/ActivityRecord;->task:Lcom/android/server/am/TaskRecord;
+ 
+     iget-object v0, v0, Lcom/android/server/am/TaskRecord;->intent:Landroid/content/Intent;
+ 
+     invoke-virtual {v0}, Landroid/content/Intent;->getComponent()Landroid/content/ComponentName;
+ 
+     move-result-object v0
+ 
+     invoke-virtual {v0}, Landroid/content/ComponentName;->flattenToShortString()Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     goto :goto_0
+ .end method
+ 
  .method private stopFullyDrawnTraceIfNeeded()V
      .locals 6
  
***************
*** 6236,6241 ****
--- 6310,6319 ----
      :cond_2
      iget-object v9, p0, Lcom/android/server/am/ActivityStack;->mTaskHistory:Ljava/util/ArrayList;
  
+     invoke-static {p0, v9, v8}, Lcom/android/server/am/ActivityStackInjector;->transferOnTopOfHomeForMoveTaskToBackLocked(Lcom/android/server/am/ActivityStack;Ljava/util/ArrayList;Lcom/android/server/am/TaskRecord;)V
+ 
+     iget-object v9, p0, Lcom/android/server/am/ActivityStack;->mTaskHistory:Ljava/util/ArrayList;
+ 
      invoke-virtual {v9, v8}, Ljava/util/ArrayList;->remove(Ljava/lang/Object;)Z
  
      iget-object v9, p0, Lcom/android/server/am/ActivityStack;->mTaskHistory:Ljava/util/ArrayList;
***************
*** 6463,6468 ****
--- 6541,6550 ----
  
      invoke-virtual {v3, v4}, Lcom/android/server/am/ActivityStackSupervisor;->moveHomeStack(Z)V
  
+     iget-object v3, p0, Lcom/android/server/am/ActivityStack;->mTaskHistory:Ljava/util/ArrayList;
+ 
+     invoke-static {p0, v3, p1}, Lcom/android/server/am/ActivityStackInjector;->transferOnTopOfHomeForMoveTaskToFrontLocked(Lcom/android/server/am/ActivityStack;Ljava/util/ArrayList;Lcom/android/server/am/TaskRecord;)V
+ 
      invoke-direct {p0, p1}, Lcom/android/server/am/ActivityStack;->insertTaskAtTop(Lcom/android/server/am/TaskRecord;)V
  
      if-eqz p2, :cond_4
***************
*** 11236,11241 ****
--- 11318,11325 ----
      if-eq v4, v6, :cond_4
  
      :cond_3
+     invoke-direct {p0, v3}, Lcom/android/server/am/ActivityStack;->startPausingLocked_Middle(Lcom/android/server/am/ActivityRecord;)V
+ 
      invoke-virtual {p0, v3}, Lcom/android/server/am/ActivityStack;->screenshotActivities(Lcom/android/server/am/ActivityRecord;)Landroid/graphics/Bitmap;
  
      move-result-object v4
