*** ./smali/com/android/server/am/BroadcastQueue.smali	2015-05-02 01:21:42.296406799 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/services.jar.out/./smali/com/android/server/am/BroadcastQueue.smali	2015-05-02 01:18:09.104413098 +0800
***************
*** 1525,1530 ****
--- 1525,1819 ----
      return-void
  .end method
  
+ .method checkApplicationAutoStart(Lcom/android/server/am/BroadcastRecord;Landroid/content/pm/ResolveInfo;)Z
+     .locals 9
+     .param p1, "r"    # Lcom/android/server/am/BroadcastRecord;
+     .param p2, "info"    # Landroid/content/pm/ResolveInfo;
+ 
+     .prologue
+     const/4 v0, 0x1
+ 
+     const/4 v6, 0x0
+ 
+     new-instance v8, Landroid/app/ActivityManager$MemoryInfo;
+ 
+     invoke-direct {v8}, Landroid/app/ActivityManager$MemoryInfo;-><init>()V
+ 
+     .local v8, "outInfo":Landroid/app/ActivityManager$MemoryInfo;
+     iget-object v1, p0, Lcom/android/server/am/BroadcastQueue;->mService:Lcom/android/server/am/ActivityManagerService;
+ 
+     invoke-virtual {v1, v8}, Lcom/android/server/am/ActivityManagerService;->getMemoryInfo(Landroid/app/ActivityManager$MemoryInfo;)V
+ 
+     iget-boolean v1, v8, Landroid/app/ActivityManager$MemoryInfo;->lowMemory:Z
+ 
+     if-eqz v1, :cond_0
+ 
+     const-string v0, "BroadcastQueue"
+ 
+     new-instance v1, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v2, "Unable to launch app "
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     iget-object v2, p2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     iget-object v2, v2, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
+ 
+     iget-object v2, v2, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     const-string v2, "/"
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     iget-object v2, p2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     iget-object v2, v2, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
+ 
+     iget v2, v2, Landroid/content/pm/ApplicationInfo;->uid:I
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     const-string v2, " for broadcast "
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     iget-object v2, p1, Lcom/android/server/am/BroadcastRecord;->intent:Landroid/content/Intent;
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     const-string v2, ": process is not permitted due to low memory"
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     invoke-static {v0, v1}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     invoke-virtual {p0, p1}, Lcom/android/server/am/BroadcastQueue;->logBroadcastReceiverDiscardLocked(Lcom/android/server/am/BroadcastRecord;)V
+ 
+     iget v2, p1, Lcom/android/server/am/BroadcastRecord;->resultCode:I
+ 
+     iget-object v3, p1, Lcom/android/server/am/BroadcastRecord;->resultData:Ljava/lang/String;
+ 
+     iget-object v4, p1, Lcom/android/server/am/BroadcastRecord;->resultExtras:Landroid/os/Bundle;
+ 
+     iget-boolean v5, p1, Lcom/android/server/am/BroadcastRecord;->resultAbort:Z
+ 
+     move-object v0, p0
+ 
+     move-object v1, p1
+ 
+     invoke-virtual/range {v0 .. v6}, Lcom/android/server/am/BroadcastQueue;->finishReceiverLocked(Lcom/android/server/am/BroadcastRecord;ILjava/lang/String;Landroid/os/Bundle;ZZ)Z
+ 
+     invoke-virtual {p0}, Lcom/android/server/am/BroadcastQueue;->scheduleBroadcastsLocked()V
+ 
+     iput v6, p1, Lcom/android/server/am/BroadcastRecord;->state:I
+ 
+     :goto_0
+     return v6
+ 
+     :cond_0
+     iget-object v1, p2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     iget-object v1, v1, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
+ 
+     iget v1, v1, Landroid/content/pm/ApplicationInfo;->flags:I
+ 
+     and-int/lit16 v1, v1, 0x81
+ 
+     if-eqz v1, :cond_2
+ 
+     sget-boolean v1, Lmiui/os/Build;->IS_STABLE_VERSION:Z
+ 
+     if-nez v1, :cond_1
+ 
+     const-string v1, "BroadcastQueue"
+ 
+     new-instance v2, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v3, "System app: \""
+ 
+     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     iget-object v3, p2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     iget-object v3, v3, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
+ 
+     iget-object v3, v3, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
+ 
+     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     const-string v3, "\" started by broadcast "
+ 
+     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     iget-object v3, p1, Lcom/android/server/am/BroadcastRecord;->intent:Landroid/content/Intent;
+ 
+     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-static {v1, v2}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     :cond_1
+     move v6, v0
+ 
+     goto :goto_0
+ 
+     :cond_2
+     iget-object v1, p1, Lcom/android/server/am/BroadcastRecord;->intent:Landroid/content/Intent;
+ 
+     invoke-virtual {v1}, Landroid/content/Intent;->getComponent()Landroid/content/ComponentName;
+ 
+     move-result-object v1
+ 
+     if-eqz v1, :cond_3
+ 
+     move v6, v0
+ 
+     goto :goto_0
+ 
+     :cond_3
+     iget-object v1, p0, Lcom/android/server/am/BroadcastQueue;->mService:Lcom/android/server/am/ActivityManagerService;
+ 
+     iget-object v1, v1, Lcom/android/server/am/ActivityManagerService;->mContext:Landroid/content/Context;
+ 
+     iget-object v2, p2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     iget-object v2, v2, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
+ 
+     iget-object v2, v2, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
+ 
+     invoke-static {v1, v2}, Landroid/miui/AppOpsUtils;->getApplicationAutoStart(Landroid/content/Context;Ljava/lang/String;)I
+ 
+     move-result v7
+ 
+     .local v7, "autoStartMode":I
+     if-nez v7, :cond_4
+ 
+     move v6, v0
+ 
+     goto :goto_0
+ 
+     :cond_4
+     const-string v0, "BroadcastQueue"
+ 
+     new-instance v1, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v2, "Unable to launch app "
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     iget-object v2, p2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     iget-object v2, v2, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
+ 
+     iget-object v2, v2, Landroid/content/pm/ApplicationInfo;->packageName:Ljava/lang/String;
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     const-string v2, "/"
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     iget-object v2, p2, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     iget-object v2, v2, Landroid/content/pm/ActivityInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
+ 
+     iget v2, v2, Landroid/content/pm/ApplicationInfo;->uid:I
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     const-string v2, " for broadcast "
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     iget-object v2, p1, Lcom/android/server/am/BroadcastRecord;->intent:Landroid/content/Intent;
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     const-string v2, ": process is not permitted to autostart"
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     invoke-static {v0, v1}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     invoke-virtual {p0, p1}, Lcom/android/server/am/BroadcastQueue;->logBroadcastReceiverDiscardLocked(Lcom/android/server/am/BroadcastRecord;)V
+ 
+     iget v2, p1, Lcom/android/server/am/BroadcastRecord;->resultCode:I
+ 
+     iget-object v3, p1, Lcom/android/server/am/BroadcastRecord;->resultData:Ljava/lang/String;
+ 
+     iget-object v4, p1, Lcom/android/server/am/BroadcastRecord;->resultExtras:Landroid/os/Bundle;
+ 
+     iget-boolean v5, p1, Lcom/android/server/am/BroadcastRecord;->resultAbort:Z
+ 
+     move-object v0, p0
+ 
+     move-object v1, p1
+ 
+     invoke-virtual/range {v0 .. v6}, Lcom/android/server/am/BroadcastQueue;->finishReceiverLocked(Lcom/android/server/am/BroadcastRecord;ILjava/lang/String;Landroid/os/Bundle;ZZ)Z
+ 
+     invoke-virtual {p0}, Lcom/android/server/am/BroadcastQueue;->scheduleBroadcastsLocked()V
+ 
+     iput v6, p1, Lcom/android/server/am/BroadcastRecord;->state:I
+ 
+     goto/16 :goto_0
+ .end method
+ 
  .method final dumpLocked(Ljava/io/FileDescriptor;Ljava/io/PrintWriter;[Ljava/lang/String;IZLjava/lang/String;Z)Z
      .locals 12
      .param p1, "fd"    # Ljava/io/FileDescriptor;
***************
*** 4820,4825 ****
--- 5109,5131 ----
      :cond_20
      move-object/from16 v0, p0
  
+     move-object/from16 v1, v32
+ 
+     move-object/from16 v2, v20
+ 
+     invoke-virtual {v0, v1, v2}, Lcom/android/server/am/BroadcastQueue;->checkApplicationAutoStart(Lcom/android/server/am/BroadcastRecord;Landroid/content/pm/ResolveInfo;)Z
+ 
+     move-result v3
+ 
+     if-nez v3, :cond_miui_0
+     
+     monitor-exit v38
+     
+     return-void
+ 
+     :cond_miui_0
+     move-object/from16 v0, p0
+ 
      iget-object v3, v0, Lcom/android/server/am/BroadcastQueue;->mService:Lcom/android/server/am/ActivityManagerService;
  
      move-object/from16 v0, v20
