*** ./smali/com/android/internal/telephony/InboundSmsHandler$SmsBroadcastReceiver.smali	2015-05-02 01:21:53.076406481 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/InboundSmsHandler$SmsBroadcastReceiver.smali	2015-05-02 01:18:20.116412773 +0800
***************
*** 55,60 ****
--- 55,80 ----
      return-void
  .end method
  
+ .method static synthetic access_mDeleteWhere(Lcom/android/internal/telephony/InboundSmsHandler$SmsBroadcastReceiver;)Ljava/lang/String;
+     .locals 1
+     .param p0, "x0"    # Lcom/android/internal/telephony/InboundSmsHandler$SmsBroadcastReceiver;
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/InboundSmsHandler$SmsBroadcastReceiver;->mDeleteWhere:Ljava/lang/String;
+ 
+     return-object v0
+ .end method
+ 
+ .method static synthetic access_mDeleteWhereArgs(Lcom/android/internal/telephony/InboundSmsHandler$SmsBroadcastReceiver;)[Ljava/lang/String;
+     .locals 1
+     .param p0, "x0"    # Lcom/android/internal/telephony/InboundSmsHandler$SmsBroadcastReceiver;
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/InboundSmsHandler$SmsBroadcastReceiver;->mDeleteWhereArgs:[Ljava/lang/String;
+ 
+     return-object v0
+ .end method
+ 
  
  # virtual methods
  .method public onReceive(Landroid/content/Context;Landroid/content/Intent;)V
