*** ./smali/com/android/internal/telephony/PhoneProxy.smali	2015-05-02 01:21:55.196406418 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/telephony-common.jar.out/./smali/com/android/internal/telephony/PhoneProxy.smali	2015-05-02 01:18:22.256412710 +0800
***************
*** 1547,1552 ****
--- 1547,1565 ----
      return v0
  .end method
  
+ .method public getMiuiIccPhoneBookInterfaceManager()Lcom/android/internal/telephony/MiuiIccPhoneBookInterfaceManager;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/PhoneProxy;->mActivePhone:Lcom/android/internal/telephony/Phone;
+ 
+     invoke-interface {v0}, Lcom/android/internal/telephony/Phone;->getMiuiIccPhoneBookInterfaceManager()Lcom/android/internal/telephony/MiuiIccPhoneBookInterfaceManager;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
  .method public getMsisdn()Ljava/lang/String;
      .locals 1
  
