*** ./smali/com/android/server/wm/WindowStateAnimator.smali	2015-05-02 01:21:36.756406963 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/services.jar.out/./smali/com/android/server/wm/WindowStateAnimator.smali	2015-05-02 01:18:03.240413272 +0800
***************
*** 2829,2834 ****
--- 2829,2849 ----
      invoke-static {}, Landroid/view/SurfaceControl;->openTransaction()V
  
      :try_start_1
+     iget v0, v7, Landroid/view/WindowManager$LayoutParams;->flags:I
+ 
+     and-int/lit8 v0, v0, 0x4
+ 
+     if-eqz v0, :cond_miui_0
+ 
+     iget-object v0, p0, Lcom/android/server/wm/WindowStateAnimator;->mSurfaceControl:Landroid/view/SurfaceControl;
+ 
+     const/16 v1, 0x80
+ 
+     const/16 v2, 0x80
+ 
+     invoke-virtual {v0, v1, v2}, Landroid/view/SurfaceControl;->setFlags(II)V
+ 
+     :cond_miui_0
      iget-object v0, p0, Lcom/android/server/wm/WindowStateAnimator;->mWin:Lcom/android/server/wm/WindowState;
  
      iget-object v0, v0, Lcom/android/server/wm/WindowState;->mFrame:Landroid/graphics/Rect;
***************
*** 5056,5061 ****
--- 5071,5082 ----
  
      if-eqz v3, :cond_2
  
+     iget v3, p0, Lcom/android/server/wm/WindowStateAnimator;->mAttrType:I
+ 
+     const/16 v4, 0x7dd
+ 
+     if-eq v3, v4, :cond_2
+ 
      :cond_1
      :goto_0
      return-void
