*** ./smali/com/android/server/wm/WindowManagerService.smali	2015-05-02 01:21:36.704406965 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/services.jar.out/./smali/com/android/server/wm/WindowManagerService.smali	2015-05-02 01:18:03.192413273 +0800
***************
*** 6054,6059 ****
--- 6054,6078 ----
  
      invoke-virtual {v0, v1}, Lcom/android/server/wm/WindowStateAnimator;->setSurfaceBoundariesLocked(Z)V
  
+     iget-object v2, v3, Lcom/android/server/wm/WindowState;->mAttrs:Landroid/view/WindowManager$LayoutParams;
+ 
+     iget v2, v2, Landroid/view/WindowManager$LayoutParams;->flags:I
+ 
+     and-int/lit8 v2, v2, 0x4
+ 
+     if-eqz v2, :cond_miui_0
+ 
+     move-object/from16 v0, v52
+ 
+     iget-object v2, v0, Lcom/android/server/wm/WindowStateAnimator;->mSurfaceControl:Landroid/view/SurfaceControl;
+ 
+     const/16 v8, 0x80
+ 
+     const/16 v9, 0x80
+ 
+     invoke-virtual {v2, v8, v9}, Landroid/view/SurfaceControl;->setFlags(II)V
+ 
+     :goto_miui_0
      iget-object v0, v3, Lcom/android/server/wm/WindowState;->mAppToken:Lcom/android/server/wm/AppWindowToken;
  
      move-object/from16 v19, v0
***************
*** 6822,6827 ****
--- 6841,6859 ----
  
      goto/16 :goto_8
  
+     :cond_miui_0
+     move-object/from16 v0, v52
+ 
+     iget-object v2, v0, Lcom/android/server/wm/WindowStateAnimator;->mSurfaceControl:Landroid/view/SurfaceControl;
+ 
+     const/4 v8, 0x0
+ 
+     const/16 v9, 0x80
+ 
+     invoke-virtual {v2, v8, v9}, Landroid/view/SurfaceControl;->setFlags(II)V
+     
+     goto :goto_miui_0
+ 
      .restart local v19    # "atoken":Lcom/android/server/wm/AppWindowToken;
      .restart local v20    # "committed":Z
      .restart local v44    # "obscuredChanged":Z
***************
*** 22056,22061 ****
--- 22088,22123 ----
      throw v1
  .end method
  
+ .method public getFocusedWindowType()I
+     .locals 2
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/wm/WindowManagerService;->mCurrentFocus:Lcom/android/server/wm/WindowState;
+ 
+     .local v0, "focus":Lcom/android/server/wm/WindowState;
+     if-eqz v0, :cond_0
+ 
+     invoke-virtual {v0}, Lcom/android/server/wm/WindowState;->getAttrs()Landroid/view/WindowManager$LayoutParams;
+ 
+     move-result-object v1
+ 
+     if-eqz v1, :cond_0
+ 
+     invoke-virtual {v0}, Lcom/android/server/wm/WindowState;->getAttrs()Landroid/view/WindowManager$LayoutParams;
+ 
+     move-result-object v1
+ 
+     iget v1, v1, Landroid/view/WindowManager$LayoutParams;->type:I
+ 
+     :goto_0
+     return v1
+ 
+     :cond_0
+     const/4 v1, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method public getInitialDisplayDensity(I)I
      .locals 4
      .param p1, "displayId"    # I
***************
*** 27027,27032 ****
--- 27089,27109 ----
      goto :goto_1
  .end method
  
+ .method public reboot()V
+     .locals 3
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/wm/WindowManagerService;->mContext:Landroid/content/Context;
+ 
+     const/4 v1, 0x0
+ 
+     const/4 v2, 0x1
+ 
+     invoke-static {v0, v1, v2}, Lcom/android/server/power/ShutdownThread;->reboot(Landroid/content/Context;Ljava/lang/String;Z)V
+ 
+     return-void
+ .end method
+ 
  .method public rebootSafeMode(Z)V
      .locals 1
      .param p1, "confirm"    # Z
