*** ./smali/com/android/server/pm/PackageManagerService$PackageHandler.smali	2015-05-02 01:21:35.864406989 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/services.jar.out/./smali/com/android/server/pm/PackageManagerService$PackageHandler.smali	2015-05-02 01:18:02.336413298 +0800
***************
*** 2303,2308 ****
--- 2303,2355 ----
  .end method
  
  .method public handleMessage(Landroid/os/Message;)V
+     .locals 3
+     .param p1, "msg"    # Landroid/os/Message;
+ 
+     .prologue
+     const/16 v2, 0xa
+ 
+     :try_start_0
+     iget-object v0, p0, Lcom/android/server/pm/PackageManagerService$PackageHandler;->this$0:Lcom/android/server/pm/PackageManagerService;
+ 
+     iget-object v0, v0, Lcom/android/server/pm/PackageManagerService;->mContext:Landroid/content/Context;
+ 
+     iget-object v1, p0, Lcom/android/server/pm/PackageManagerService$PackageHandler;->this$0:Lcom/android/server/pm/PackageManagerService;
+ 
+     iget-boolean v1, v1, Lcom/android/server/pm/PackageManagerService;->mSystemReady:Z
+ 
+     invoke-static {v0, p1, v1}, Lcom/android/server/pm/MiuiPackageManagerService;->checkApk(Landroid/content/Context;Landroid/os/Message;Z)Z
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_0
+ 
+     invoke-static {v2}, Landroid/os/Process;->setThreadPriority(I)V
+ 
+     :goto_0
+     return-void
+ 
+     :cond_0
+     :try_start_1
+     invoke-virtual {p0, p1}, Lcom/android/server/pm/PackageManagerService$PackageHandler;->doHandleMessage(Landroid/os/Message;)V
+     :try_end_1
+     .catchall {:try_start_1 .. :try_end_1} :catchall_0
+ 
+     invoke-static {v2}, Landroid/os/Process;->setThreadPriority(I)V
+ 
+     goto :goto_0
+ 
+     :catchall_0
+     move-exception v0
+ 
+     invoke-static {v2}, Landroid/os/Process;->setThreadPriority(I)V
+ 
+     throw v0
+ .end method
+ 
+ .method public handleMessage_aosp(Landroid/os/Message;)V
      .locals 2
      .param p1, "msg"    # Landroid/os/Message;
  
