*** ./smali/android/content/res/Configuration.smali	2015-05-02 01:20:38.708408678 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework.jar.out/./smali/android/content/res/Configuration.smali	2015-05-02 01:17:00.652415121 +0800
***************
*** 200,205 ****
--- 200,207 ----
  
  .field public densityDpi:I
  
+ .field public extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
  .field public fontScale:F
  
  .field public hardKeyboardHidden:I
***************
*** 258,292 ****
  .end method
  
  .method public constructor <init>()V
!     .locals 0
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0}, Landroid/content/res/Configuration;->setToDefaults()V
  
      return-void
  .end method
  
  .method public constructor <init>(Landroid/content/res/Configuration;)V
!     .locals 0
      .param p1, "o"    # Landroid/content/res/Configuration;
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->setTo(Landroid/content/res/Configuration;)V
  
      return-void
  .end method
  
  .method private constructor <init>(Landroid/os/Parcel;)V
!     .locals 0
      .param p1, "source"    # Landroid/os/Parcel;
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->readFromParcel(Landroid/os/Parcel;)V
  
      return-void
--- 260,312 ----
  .end method
  
  .method public constructor <init>()V
!     .locals 1
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-direct {v0}, Landroid/content/res/MiuiConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
      invoke-virtual {p0}, Landroid/content/res/Configuration;->setToDefaults()V
  
      return-void
  .end method
  
  .method public constructor <init>(Landroid/content/res/Configuration;)V
!     .locals 1
      .param p1, "o"    # Landroid/content/res/Configuration;
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-direct {v0}, Landroid/content/res/MiuiConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->setTo(Landroid/content/res/Configuration;)V
  
      return-void
  .end method
  
  .method private constructor <init>(Landroid/os/Parcel;)V
!     .locals 1
      .param p1, "source"    # Landroid/os/Parcel;
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-direct {v0}, Landroid/content/res/MiuiConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->readFromParcel(Landroid/os/Parcel;)V
  
      return-void
***************
*** 325,338 ****
  
      and-int/2addr v0, p0
  
!     if-eqz v0, :cond_0
  
      const/4 v0, 0x1
  
      :goto_0
      return v0
  
!     :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
--- 345,365 ----
  
      and-int/2addr v0, p0
  
!     if-nez v0, :cond_0
  
+     invoke-static {p0}, Landroid/content/res/MiuiConfiguration;->needNewResources(I)Z
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_1
+ 
+     :cond_0
      const/4 v0, 0x1
  
      :goto_0
      return v0
  
!     :cond_1
      const/4 v0, 0x0
  
      goto :goto_0
***************
*** 712,717 ****
--- 739,752 ----
  
      sub-int v2, v3, v4
  
+     iget-object v3, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     iget-object v4, p1, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-virtual {v3, v4}, Landroid/content/res/MiuiConfiguration;->compareTo(Landroid/content/res/MiuiConfiguration;)I
+ 
+     move-result v2
+ 
      goto/16 :goto_0
  .end method
  
***************
*** 1015,1020 ****
--- 1050,1065 ----
      or-int/lit16 v0, v0, 0x1000
  
      :cond_12
+     iget-object v2, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     iget-object v3, p1, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-virtual {v2, v3}, Landroid/content/res/MiuiConfiguration;->diff(Landroid/content/res/MiuiConfiguration;)I
+ 
+     move-result v2
+ 
+     or-int/2addr v0, v2
+ 
      return v0
  .end method
  
***************
*** 1218,1223 ****
--- 1263,1278 ----
  
      add-int v0, v1, v2
  
+     mul-int/lit8 v1, v0, 0x1f
+ 
+     iget-object v2, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-virtual {v2}, Landroid/content/res/MiuiConfiguration;->hashCode()I
+ 
+     move-result v2
+ 
+     add-int v0, v1, v2
+ 
      return v0
  
      :cond_0
***************
*** 1473,1478 ****
--- 1528,1537 ----
  
      iput v0, p0, Landroid/content/res/Configuration;->seq:I
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-virtual {v0, p1}, Landroid/content/res/MiuiConfiguration;->readFromParcel(Landroid/os/Parcel;)V
+ 
      return-void
  
      :cond_1
***************
*** 1525,1531 ****
  .end method
  
  .method public setTo(Landroid/content/res/Configuration;)V
!     .locals 1
      .param p1, "o"    # Landroid/content/res/Configuration;
  
      .prologue
--- 1584,1590 ----
  .end method
  
  .method public setTo(Landroid/content/res/Configuration;)V
!     .locals 2
      .param p1, "o"    # Landroid/content/res/Configuration;
  
      .prologue
***************
*** 1628,1633 ****
--- 1687,1698 ----
  
      iput v0, p0, Landroid/content/res/Configuration;->seq:I
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     iget-object v1, p1, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-virtual {v0, v1}, Landroid/content/res/MiuiConfiguration;->setTo(Landroid/content/res/MiuiConfiguration;)V
+ 
      return-void
  .end method
  
***************
*** 1685,1690 ****
--- 1750,1759 ----
  
      iput v1, p0, Landroid/content/res/Configuration;->seq:I
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-virtual {v0}, Landroid/content/res/MiuiConfiguration;->setToDefaults()V
+ 
      return-void
  .end method
  
***************
*** 2009,2014 ****
--- 2078,2091 ----
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
      :cond_0
+     iget-object v2, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-virtual {v2}, Landroid/content/res/MiuiConfiguration;->toString()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
      const/16 v2, 0x7d
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
***************
*** 2937,2942 ****
--- 3014,3029 ----
      iput v2, p0, Landroid/content/res/Configuration;->seq:I
  
      :cond_19
+     iget-object v2, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     iget-object v3, p1, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-virtual {v2, v3}, Landroid/content/res/MiuiConfiguration;->updateFrom(Landroid/content/res/MiuiConfiguration;)I
+ 
+     move-result v2
+ 
+     or-int/2addr v0, v2
+ 
      return v0
  
      .end local v1    # "deltaScreenLayoutDir":I
***************
*** 3058,3063 ****
--- 3145,3154 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Landroid/content/res/MiuiConfiguration;
+ 
+     invoke-virtual {v0, p1, p2}, Landroid/content/res/MiuiConfiguration;->writeToParcel(Landroid/os/Parcel;I)V
+ 
      return-void
  
      :cond_0
