*** ./smali/android/content/res/Resources$Theme.smali	2015-05-02 01:20:38.696408679 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework.jar.out/./smali/android/content/res/Resources$Theme.smali	2015-05-02 01:17:00.632415122 +0800
***************
*** 135,141 ****
  
      invoke-static/range {v0 .. v6}, Landroid/content/res/AssetManager;->applyStyle(IIII[I[I[I)Z
  
!     return-object v7
  .end method
  
  .method public obtainStyledAttributes(Landroid/util/AttributeSet;[III)Landroid/content/res/TypedArray;
--- 135,147 ----
  
      invoke-static/range {v0 .. v6}, Landroid/content/res/AssetManager;->applyStyle(IIII[I[I[I)Z
  
!     iget-object v0, p0, Landroid/content/res/Resources$Theme;->this$0:Landroid/content/res/Resources;
! 
!     invoke-virtual {v0, v7}, Landroid/content/res/Resources;->loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
! 
!     move-result-object v0
! 
!     return-object v0
  .end method
  
  .method public obtainStyledAttributes(Landroid/util/AttributeSet;[III)Landroid/content/res/TypedArray;
***************
*** 185,191 ****
  
      iput-object v9, v7, Landroid/content/res/TypedArray;->mXml:Landroid/content/res/XmlBlock$Parser;
  
!     return-object v7
  
      :cond_0
      const/4 v3, 0x0
--- 191,203 ----
  
      iput-object v9, v7, Landroid/content/res/TypedArray;->mXml:Landroid/content/res/XmlBlock$Parser;
  
!     iget-object v0, p0, Landroid/content/res/Resources$Theme;->this$0:Landroid/content/res/Resources;
! 
!     invoke-virtual {v0, v7}, Landroid/content/res/Resources;->loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
! 
!     move-result-object v0
! 
!     return-object v0
  
      :cond_0
      const/4 v3, 0x0
***************
*** 227,233 ****
  
      invoke-static/range {v0 .. v6}, Landroid/content/res/AssetManager;->applyStyle(IIII[I[I[I)Z
  
!     return-object v7
  .end method
  
  .method public resolveAttribute(ILandroid/util/TypedValue;Z)Z
--- 239,251 ----
  
      invoke-static/range {v0 .. v6}, Landroid/content/res/AssetManager;->applyStyle(IIII[I[I[I)Z
  
!     iget-object v0, p0, Landroid/content/res/Resources$Theme;->this$0:Landroid/content/res/Resources;
! 
!     invoke-virtual {v0, v7}, Landroid/content/res/Resources;->loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
! 
!     move-result-object v0
! 
!     return-object v0
  .end method
  
  .method public resolveAttribute(ILandroid/util/TypedValue;Z)Z
