*** ./smali/android/net/IConnectivityManager$Stub.smali	2015-05-02 01:21:03.256407953 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework.jar.out/./smali/android/net/IConnectivityManager$Stub.smali	2015-05-02 01:17:27.080414340 +0800
***************
*** 144,149 ****
--- 144,152 ----
  
  .field static final TRANSACTION_updateLockdownVpn:I = 0x2f
  
+ .field static final TRANSACTION_startUsingNetworkFeatureMSim:I = 0x3c
+ 
+ .field static final TRANSACTION_stopUsingNetworkFeatureMSim:I = 0x3d
  
  # direct methods
  .method public constructor <init>()V
***************
*** 1888,1893 ****
--- 1891,1963 ----
  
      goto :goto_11
  
+     :sswitch_miui_0
+     const-string v5, "android.net.IConnectivityManager"
+ 
+     invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
+ 
+     invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v0
+ 
+     .restart local v0    # "_arg0":I
+     invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     .restart local v1    # "_arg1":Ljava/lang/String;
+     invoke-virtual {p2}, Landroid/os/Parcel;->readStrongBinder()Landroid/os/IBinder;
+ 
+     move-result-object v2
+ 
+     .restart local v2    # "_arg2":Landroid/os/IBinder;
+     invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v3
+ 
+     .local v3, "_arg3":I
+     invoke-virtual {p0, v0, v1, v2, v3}, Landroid/net/IConnectivityManager$Stub;->startUsingNetworkFeatureMSim(ILjava/lang/String;Landroid/os/IBinder;I)I
+ 
+     move-result v4
+ 
+     .restart local v4    # "_result":I
+     invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
+ 
+     invoke-virtual {p3, v4}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     goto/16 :goto_0
+ 
+     :sswitch_miui_1
+     const-string v5, "android.net.IConnectivityManager"
+ 
+     invoke-virtual {p2, v5}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
+ 
+     invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v0
+ 
+     .restart local v0    # "_arg0":I
+     invoke-virtual {p2}, Landroid/os/Parcel;->readString()Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     .restart local v1    # "_arg1":Ljava/lang/String;
+     invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v2
+ 
+     .local v2, "_arg2":I
+     invoke-virtual {p0, v0, v1, v2}, Landroid/net/IConnectivityManager$Stub;->stopUsingNetworkFeatureMSim(ILjava/lang/String;I)I
+ 
+     move-result v4
+ 
+     .restart local v4    # "_result":I
+     invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
+ 
+     invoke-virtual {p3, v4}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     goto/16 :goto_0
+ 
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_1
***************
*** 1949,1954 ****
--- 2019,2026 ----
          0x39 -> :sswitch_39
          0x3a -> :sswitch_3a
          0x3b -> :sswitch_3b
+         0x3c -> :sswitch_miui_0
+         0x3d -> :sswitch_miui_1
          0x5f4e5446 -> :sswitch_0
      .end sparse-switch
  .end method
