*** ./smali/com/android/server/ConnectivityService$CheckMp.smali	2015-05-02 01:21:43.416406766 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/services.jar.out/./smali/com/android/server/ConnectivityService$CheckMp.smali	2015-05-02 01:18:10.220413066 +0800
***************
*** 2130,2135 ****
--- 2130,2151 ----
      goto/16 :goto_0
  
      :cond_1f
+     const/16 v35, 0xc8
+ 
+     move/from16 v0, v28
+ 
+     move/from16 v1, v35
+ 
+     if-lt v0, v1, :cond_miui_0
+ 
+     const/16 v35, 0x18f
+ 
+     move/from16 v0, v28
+ 
+     move/from16 v1, v35
+ 
+     if-gt v0, v1, :cond_miui_0
+ 
      :try_start_18
      new-instance v35, Ljava/lang/StringBuilder;
  
***************
*** 2164,2169 ****
--- 2180,2186 ----
  
      move-result-object v29
  
+     :cond_miui_0
      move v6, v5
  
      .end local v5    # "addrTried":I
