*** ./smali/com/android/server/LockSettingsService.smali	2015-05-02 01:21:35.756406993 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/services.jar.out/./smali/com/android/server/LockSettingsService.smali	2015-05-02 01:18:02.212413302 +0800
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/server/LockSettingsService$DatabaseHelper;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/server/LockSettingsService$DatabaseHelper;,
!         Lcom/android/server/LockSettingsService$Injector;
      }
  .end annotation
  
***************
*** 1259,1265 ****
      move-result-object v2
  
      .local v2, "hash":[B
!     invoke-static {v6, v2}, Ljava/util/Arrays;->equals([B[B)Z
  
      move-result v4
  
--- 1260,1270 ----
      move-result-object v2
  
      .local v2, "hash":[B
!     invoke-static {v6, v2}, Lcom/android/server/LockSettingsService$Injector;->passwordToHash([B[B)[B
! 
!     move-result-object v8
! 
!     invoke-static {v6, v8}, Ljava/util/Arrays;->equals([B[B)Z
  
      move-result v4
  
***************
*** 1916,1921 ****
--- 1921,1948 ----
  
      return-void
  .end method
+ 
+ .method public setRawLockPassword([BI)V
+     .locals 1
+     .param p1, "hash"    # [B
+     .param p2, "userId"    # I
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ 
+     .prologue
+     invoke-direct {p0, p2}, Lcom/android/server/LockSettingsService;->checkWritePermission(I)V
+ 
+     invoke-direct {p0, p2}, Lcom/android/server/LockSettingsService;->getLockPasswordFilename(I)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     invoke-direct {p0, v0, p1}, Lcom/android/server/LockSettingsService;->writeFile(Ljava/lang/String;[B)V
+ 
+     return-void
+ .end method
  
  .method public setString(Ljava/lang/String;Ljava/lang/String;I)V
      .locals 0
