*** ./smali/android/net/wifi/WifiStateMachine.smali	2015-05-02 01:21:05.332407892 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiStateMachine.smali	2015-05-02 01:17:29.352414273 +0800
***************
*** 936,942 ****
  
      iget-object v8, p0, Landroid/net/wifi/WifiStateMachine;->mWifiNative:Landroid/net/wifi/WifiNative;
  
!     invoke-direct {v7, p1, v8}, Landroid/net/wifi/WifiConfigStore;-><init>(Landroid/content/Context;Landroid/net/wifi/WifiNative;)V
  
      iput-object v7, p0, Landroid/net/wifi/WifiStateMachine;->mWifiConfigStore:Landroid/net/wifi/WifiConfigStore;
  
--- 936,946 ----
  
      iget-object v8, p0, Landroid/net/wifi/WifiStateMachine;->mWifiNative:Landroid/net/wifi/WifiNative;
  
!     invoke-virtual {p0}, Landroid/net/wifi/WifiStateMachine;->getHandler()Landroid/os/Handler;
! 
!     move-result-object v9
! 
!     invoke-direct {v7, p1, v8, v9}, Landroid/net/wifi/WifiConfigStore;-><init>(Landroid/content/Context;Landroid/net/wifi/WifiNative;Landroid/os/Handler;)V
  
      iput-object v7, p0, Landroid/net/wifi/WifiStateMachine;->mWifiConfigStore:Landroid/net/wifi/WifiConfigStore;
  
***************
*** 5027,5032 ****
--- 5031,5040 ----
  
      invoke-virtual {v1, v4}, Landroid/net/wifi/WifiInfo;->setMeteredHint(Z)V
  
+     iget-object v1, p0, Landroid/net/wifi/WifiStateMachine;->mWifiInfo:Landroid/net/wifi/WifiInfo;
+ 
+     invoke-virtual {v1, v2}, Landroid/net/wifi/WifiInfo;->setVendorInfo(Ljava/lang/String;)V
+ 
      invoke-direct {p0}, Landroid/net/wifi/WifiStateMachine;->getMaxDhcpRetries()I
  
      move-result v0
***************
*** 5156,5161 ****
--- 5164,5173 ----
  
      invoke-virtual {v1, v2}, Landroid/net/wifi/WifiInfo;->setMeteredHint(Z)V
  
+     iget-object v1, p0, Landroid/net/wifi/WifiStateMachine;->mWifiInfo:Landroid/net/wifi/WifiInfo;
+ 
+     invoke-virtual {v1, v4}, Landroid/net/wifi/WifiInfo;->setVendorInfo(Ljava/lang/String;)V
+ 
      sget-object v1, Landroid/net/NetworkInfo$DetailedState;->DISCONNECTED:Landroid/net/NetworkInfo$DetailedState;
  
      invoke-direct {p0, v1}, Landroid/net/wifi/WifiStateMachine;->setNetworkDetailedState(Landroid/net/NetworkInfo$DetailedState;)V
***************
*** 5350,5355 ****
--- 5362,5373 ----
  
      invoke-virtual {v3, v4}, Landroid/net/wifi/WifiInfo;->setMeteredHint(Z)V
  
+     iget-object v3, p0, Landroid/net/wifi/WifiStateMachine;->mWifiInfo:Landroid/net/wifi/WifiInfo;
+ 
+     iget-object v4, p1, Landroid/net/DhcpResults;->vendorInfo:Ljava/lang/String;
+ 
+     invoke-virtual {v3, v4}, Landroid/net/wifi/WifiInfo;->setVendorInfo(Ljava/lang/String;)V
+ 
      invoke-direct {p0}, Landroid/net/wifi/WifiStateMachine;->updateLinkProperties()V
  
      return-void
***************
*** 5419,5430 ****
--- 5437,5466 ----
      invoke-virtual {v2, v3}, Landroid/net/wifi/WifiInfo;->setNetworkId(I)V
  
      :goto_0
+     sget-object v2, Landroid/net/wifi/SupplicantState;->ASSOCIATING:Landroid/net/wifi/SupplicantState;
+ 
+     if-eq v0, v2, :cond_miui_0
+ 
+     sget-object v2, Landroid/net/wifi/SupplicantState;->ASSOCIATED:Landroid/net/wifi/SupplicantState;
+ 
+     if-eq v0, v2, :cond_miui_0
+ 
+     sget-object v2, Landroid/net/wifi/SupplicantState;->FOUR_WAY_HANDSHAKE:Landroid/net/wifi/SupplicantState;
+ 
+     if-ne v0, v2, :cond_miui_1
+ 
+     :cond_miui_0
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine;->mWifiInfo:Landroid/net/wifi/WifiInfo;
  
      iget-object v3, v1, Landroid/net/wifi/StateChangeResult;->BSSID:Ljava/lang/String;
  
      invoke-virtual {v2, v3}, Landroid/net/wifi/WifiInfo;->setBSSID(Ljava/lang/String;)V
  
+     iget-object v2, p0, Landroid/net/wifi/WifiStateMachine;->mInterfaceName:Ljava/lang/String;
+ 
+     invoke-static {v2}, Landroid/net/NetworkUtils;->enableInterface(Ljava/lang/String;)I
+ 
+     :cond_miui_1
      iget-object v2, p0, Landroid/net/wifi/WifiStateMachine;->mWifiInfo:Landroid/net/wifi/WifiInfo;
  
      iget-object v3, v1, Landroid/net/wifi/StateChangeResult;->wifiSsid:Landroid/net/wifi/WifiSsid;
***************
*** 7411,7417 ****
  .end method
  
  .method private setScanResults()V
!     .locals 30
  
      .prologue
      const-string v4, ""
--- 7447,7453 ----
  .end method
  
  .method private setScanResults()V
!     .locals 38
  
      .prologue
      const-string v4, ""
***************
*** 7440,7445 ****
--- 7476,7487 ----
      const/16 v24, 0x0
  
      .local v24, "sid":I
+     const/16 v30, 0x0
+ 
+     .local v30, "isWpsConfigured":Z
+     const/16 v31, 0x0
+ 
+     .local v31, "isXiaomiRouter":Z
      :cond_0
      move-object/from16 v0, p0
  
***************
*** 7623,7628 ****
--- 7665,7684 ----
      move-result v14
  
      .local v14, "flagLen":I
+     const-string v32, "wps_state="
+ 
+     invoke-virtual/range {v32 .. v32}, Ljava/lang/String;->length()I
+ 
+     move-result v32
+ 
+     .local v32, "wpsStateStrLength":I
+     const-string v33, "wps_device_name="
+ 
+     invoke-virtual/range {v33 .. v33}, Ljava/lang/String;->length()I
+ 
+     move-result v33
+ 
+     .local v33, "wpsDeviceNameStrLength":I
      move-object/from16 v10, v21
  
      .local v10, "arr$":[Ljava/lang/String;
***************
*** 7932,7938 ****
  
      move-result v27
  
!     if-eqz v27, :cond_6
  
      :cond_d
      if-eqz v4, :cond_e
--- 7988,7994 ----
  
      move-result v27
  
!     if-eqz v27, :cond_miui_1
  
      :cond_d
      if-eqz v4, :cond_e
***************
*** 8009,8014 ****
--- 8065,8078 ----
      iput-wide v8, v2, Landroid/net/wifi/ScanResult;->timestamp:J
  
      :goto_8
+     move/from16 v0, v30
+ 
+     iput-boolean v0, v2, Landroid/net/wifi/ScanResult;->isWpsConfigured:Z
+ 
+     move/from16 v0, v31
+ 
+     iput-boolean v0, v2, Landroid/net/wifi/ScanResult;->isXiaomiRouter:Z
+ 
      move-object/from16 v0, p0
  
      iget-object v0, v0, Landroid/net/wifi/WifiStateMachine;->mScanResults:Ljava/util/List;
***************
*** 8035,8040 ****
--- 8099,8108 ----
  
      const/4 v3, 0x0
  
+     const/16 v30, 0x0
+ 
+     const/16 v31, 0x0
+ 
      goto/16 :goto_5
  
      :cond_f
***************
*** 8096,8103 ****
      .restart local v17    # "i$":I
      .restart local v19    # "len$":I
      .restart local v21    # "lines":[Ljava/lang/String;
!     :cond_12
      :try_start_8
      monitor-exit v28
      :try_end_8
      .catchall {:try_start_8 .. :try_end_8} :catchall_0
--- 8164,8279 ----
      .restart local v17    # "i$":I
      .restart local v19    # "len$":I
      .restart local v21    # "lines":[Ljava/lang/String;
!     :cond_miui_1
      :try_start_8
+     const-string v34, "wps_state="
+ 
+     move-object/from16 v0, v20
+ 
+     move-object/from16 v1, v34
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
+ 
+     move-result v34
+ 
+     if-eqz v34, :cond_miui_0
+ 
+     new-instance v35, Ljava/lang/String;
+ 
+     invoke-virtual/range {v20 .. v20}, Ljava/lang/String;->getBytes()[B
+ 
+     move-result-object v34
+ 
+     invoke-virtual/range {v20 .. v20}, Ljava/lang/String;->length()I
+ 
+     move-result v36
+ 
+     sub-int v36, v36, v32
+ 
+     move-object/from16 v0, v35
+ 
+     move-object/from16 v1, v34
+ 
+     move-object/from16 v37, v3
+     
+     move/from16 v2, v32
+ 
+     move/from16 v3, v36
+ 
+     invoke-direct {v0, v1, v2, v3}, Ljava/lang/String;-><init>([BII)V
+ 
+     .local v35, "wpsStateStr":Ljava/lang/String;
+ 
+     move-object/from16 v3, v37
+ 
+     const-string v34, "configured"
+ 
+     move-object/from16 v0, v34
+ 
+     move-object/from16 v1, v35
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v30
+ 
+     goto/16 :goto_5
+ 
+     .end local v35    # "wpsStateStr":Ljava/lang/String;
+     :cond_miui_0
+     const-string v34, "wps_device_name="
+ 
+     move-object/from16 v0, v20
+ 
+     move-object/from16 v1, v34
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
+ 
+     move-result v34
+ 
+     if-eqz v34, :cond_6
+ 
+     new-instance v35, Ljava/lang/String;
+ 
+     invoke-virtual/range {v20 .. v20}, Ljava/lang/String;->getBytes()[B
+ 
+     move-result-object v34
+ 
+     invoke-virtual/range {v20 .. v20}, Ljava/lang/String;->length()I
+ 
+     move-result v36
+ 
+     sub-int v36, v36, v33
+ 
+     move-object/from16 v0, v35
+ 
+     move-object/from16 v1, v34
+ 
+     move-object/from16 v37, v3
+ 
+     move/from16 v2, v33
+ 
+     move/from16 v3, v36
+ 
+     invoke-direct {v0, v1, v2, v3}, Ljava/lang/String;-><init>([BII)V
+ 
+     .local v35, "wpsDeviceNameStr":Ljava/lang/String;
+ 
+     move-object/from16 v3, v37
+ 
+     const-string v34, "XiaoMiRouter"
+ 
+     move-object/from16 v0, v34
+ 
+     move-object/from16 v1, v35
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v31
+ 
+     goto/16 :goto_5
+ 
+     .end local v35    # "wpsDeviceNameStr":Ljava/lang/String;
+     :cond_12
      monitor-exit v28
      :try_end_8
      .catchall {:try_start_8 .. :try_end_8} :catchall_0
***************
*** 9954,9959 ****
--- 10130,10148 ----
      .param p2, "workSource"    # Landroid/os/WorkSource;
  
      .prologue
+     iget-object v0, p0, Landroid/net/wifi/WifiStateMachine;->mContext:Landroid/content/Context;
+ 
+     iget-object v1, p0, Landroid/net/wifi/WifiStateMachine;->mP2pConnected:Ljava/util/concurrent/atomic/AtomicBoolean;
+ 
+     invoke-static {v0, v1}, Landroid/net/wifi/WifiStateMachineInjector;->cancelScan(Landroid/content/Context;Ljava/util/concurrent/atomic/AtomicBoolean;)Z
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_miui_0
+ 
+     return-void
+ 
+     :cond_miui_0
      const v0, 0x20047
  
      const/4 v1, 0x0
