*** ./smali/android/net/wifi/WifiConfiguration.smali	2015-05-02 01:21:05.468407888 +0800
--- /home/hgm/MIUI6/8675-FHD/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiConfiguration.smali	2015-05-02 01:17:29.556414267 +0800
***************
*** 97,102 ****
--- 97,112 ----
  
  .field public status:I
  
+ .field public wapiASCert:Ljava/lang/String;
+ 
+ .field public wapiCertIndex:I
+ 
+ .field public wapiPsk:Ljava/lang/String;
+ 
+ .field public wapiPskType:I
+ 
+ .field public wapiUserCert:Ljava/lang/String;
+ 
  .field public wepKeys:[Ljava/lang/String;
  
  .field public wepTxKeyIndex:I
***************
*** 225,230 ****
--- 235,252 ----
      goto :goto_0
  
      :cond_0
+     const/4 v1, 0x0
+ 
+     iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiASCert:Ljava/lang/String;
+ 
+     iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiUserCert:Ljava/lang/String;
+ 
+     const/4 v1, -0x1
+     
+     iput v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiCertIndex:I
+ 
+     iput v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiPskType:I
+ 
      new-instance v1, Landroid/net/wifi/WifiEnterpriseConfig;
  
      invoke-direct {v1}, Landroid/net/wifi/WifiEnterpriseConfig;-><init>()V
***************
*** 281,286 ****
--- 303,320 ----
  
      iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->preSharedKey:Ljava/lang/String;
  
+     const/4 v1, 0x0
+     
+     iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiASCert:Ljava/lang/String;
+ 
+     iput-object v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiUserCert:Ljava/lang/String;
+ 
+     const/4 v1, -0x1
+ 
+     iput v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiCertIndex:I
+ 
+     iput v1, p0, Landroid/net/wifi/WifiConfiguration;->wapiPskType:I
+ 
      const/4 v1, 0x4
  
      new-array v1, v1, [Ljava/lang/String;
***************
*** 518,526 ****
  .end method
  
  .method public getAuthType()I
!     .locals 5
  
      .prologue
      const/4 v1, 0x4
  
      const/4 v3, 0x3
--- 552,562 ----
  .end method
  
  .method public getAuthType()I
!     .locals 6
  
      .prologue
+     const/4 v5, 0x5
+ 
      const/4 v1, 0x4
  
      const/4 v3, 0x3
***************
*** 595,600 ****
--- 631,664 ----
      goto :goto_0
  
      :cond_4
+     iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+ 
+     invoke-virtual {v0, v4}, Ljava/util/BitSet;->get(I)Z
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_miui_0
+ 
+     move v0, v5
+ 
+     goto :goto_0
+ 
+     :cond_miui_0
+     iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->allowedKeyManagement:Ljava/util/BitSet;
+ 
+     const/4 v1, 0x6
+ 
+     invoke-virtual {v0, v1}, Ljava/util/BitSet;->get(I)Z
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_miui_1
+ 
+     const/4 v0, 0x6
+ 
+     goto :goto_0
+ 
+     :cond_miui_1
      const/4 v0, 0x0
  
      goto :goto_0
***************
*** 1466,1471 ****
--- 1530,1551 ----
  
      invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     iget-object v4, p0, Landroid/net/wifi/WifiConfiguration;->wapiASCert:Ljava/lang/String;
+ 
+     invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
+ 
+     iget-object v4, p0, Landroid/net/wifi/WifiConfiguration;->wapiUserCert:Ljava/lang/String;
+ 
+     invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
+ 
+     iget-object v4, p0, Landroid/net/wifi/WifiConfiguration;->wapiPsk:Ljava/lang/String;
+ 
+     invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
+ 
+     iget v4, p0, Landroid/net/wifi/WifiConfiguration;->wapiPskType:I
+ 
+     invoke-virtual {p1, v4}, Landroid/os/Parcel;->writeInt(I)V
+ 
      iget-object v0, p0, Landroid/net/wifi/WifiConfiguration;->wepKeys:[Ljava/lang/String;
  
      .local v0, "arr$":[Ljava/lang/String;
